lib/netconf/provider.go

```diff
@@ -81,6 +81,57 @@ func AwaitOnChain(ctx context.Context, netID ID, portalRegistry *bindings.Portal
 	}
 }
 
+// CheckOnChain returns the network configuration if all expected chains are registered in the on-chain registry.
+// It returns an error if the context is canceled or if any of the expected chains are missing.
+func CheckOnChain(ctx context.Context, netID ID, portalRegistry *bindings.PortalRegistry, expected []string) (Network, error) {
+	if netID == Simnet {
```

Comment:
Do we need this for this func? If yes - maybe just move this dummy simnet config into one place and use it here and in AwaitOnChain?

Reply:
Good one, no this would only be called for devnet.